lib/charms/prometheus_k8s/v0/prometheus.py

```diff
@@ -573,48 +579,91 @@ def _labeled_static_job_config(self, job, job_name_prefix, hosts, scrape_metadat
         name = job.get("job_name")
         job_name = "{}_{}".format(job_name_prefix, name) if name else job_name_prefix
 
-        config = {"job_name": job_name, "metrics_path": job["metrics_path"]}
+        original_static_configs = job.get("static_configs")
+        static_configs = []
 
-        static_configs = job.get("static_configs")
-        config["static_configs"] = []
+        for static_config in original_static_configs:
+            labels = static_config.get("labels", {})
+            targets = static_config.get("targets", [])
 
-        relabel_config = {
-            "source_labels": ["juju_model", "juju_model_uuid", "juju_application"],
-            "separator": "_",
-            "target_label": "instance",
-            "regex": "(.*)",
-        }
-
-        for static_config in static_configs:
-            labels = static_config.get("labels", {}) if static_configs else {}
-            all_targets = static_config.get("targets", [])
-
-            ports = []
-            unitless_targets = []
-            for target in all_targets:
-                host, port = target.split(":")
-                if host.strip() == "*":
-                    ports.append(port.strip())
-                else:
-                    unitless_targets.append(target)
-
-            if unitless_targets:
-                unitless_config = self._labeled_unitless_config(
-                    unitless_targets, labels, scrape_metadata
+            if unitless_targets := [
+                target for target in targets if not target.startswith("*:")
+            ]:
+                static_configs.append(
+                    self._labeled_unitless_config(
+                        unitless_targets, labels, scrape_metadata
+                    )
                 )
-                config["static_configs"].append(unitless_config)
 
-            for host_name, host_address in hosts.items():
-                static_config = self._labeled_unit_config(
-                    host_name, host_address, ports, labels, scrape_metadata
-                )
-                config["static_configs"].append(static_config)
-                if "juju_unit" not in relabel_config["source_labels"]:
-                    relabel_config["source_labels"].append("juju_unit")
+            if ports := [
```

Comment:
I do not think this change is warranted either. Nesting a conditional list comprehension **in** a walrus assignment expression **in** a if condition (that is immediately followed by another if condition which could have been a conjunction),  does not improve readability and any miniscule performance gains are not required here.